### PR TITLE
Update salsa

### PR DESCRIPTION
**Pull request by jonas-schievink**
This updates salsa to include https://github.com/salsa-rs/salsa/pull/265, and removes all cancellation-related code from rust-analyzer

### CODE REVIEW

File: crates/base_db/src/cancellation.rs [hunk 1]
```diff
@@ -1,48 +0,0 @@
...
-
-impl Canceled {
-    pub(crate) fn new() -> Canceled {
-        Canceled { _private: () }
-    }
-
-    pub fn throw() -> ! {
-        // We use resume and not panic here to avoid running the panic
-        // hook (that is, to avoid collecting and printing backtrace).
-        std::panic::resume_unwind(Box::new(Canceled::new()))
```
Comment (matklad) [inline crates/base_db/src/cancellation.rs:32]: Might want to port this comment over to salsa.

File: crates/base_db/src/lib.rs [hunk 2]
```diff
@@ -38,45 +36,6 @@ pub trait Upcast<T: ?Sized> {
...
-    /// code while the code is being modified. A common problem is that a
-    /// long-running query is being calculated when a new change arrives.
-    ///
-    /// We can't just apply the change immediately: this will cause the pending
-    /// query to see inconsistent state (it will observe an absence of
-    /// repeatable read). So what we do is we **cancel** all pending queries
-    /// before applying the change.
-    ///
-    /// We implement cancellation by panicking with a special value and catching
-    /// it on the API boundary. Salsa explicitly supports this use-case.
```
Comment (matklad) [inline crates/base_db/src/lib.rs:54]: Let's move this architectural comment over to Analysis::with_db

**Comment by jonas-schievink** [comment]
Updated to salsa master, now waiting on a new salsa version to be published

**Comment by matklad** [comment]
Let's learn from our mistakes and merge on Monday :)

**Pull request merged**